File: mall-goods-interface/src/main/java/com/mall/goods/service/TypeTemplateService.java
package com.mall.goods.service;

import com.mall.pojo.TypeTemplate;
import common.pojo.PageResult;

import java.util.List;

/**
 * 商品类型模板服务接口
 * @author Wwl
 *
 */
public interface TypeTemplateService {

	/**
	 * 增加
	 * @param typeTemplate
	 */
	public void add(TypeTemplate typeTemplate);

	/**
	 * 修改
	 * @param typeTemplate
	 */
	public void update(TypeTemplate typeTemplate);

	/**
	 * 根据Id获取实体
	 * @param id
	 * */
	public TypeTemplate findOne(Long id);

	/**
	 * 查询所有
	 * @return
	 */
	public List<TypeTemplate> findAll();
	
	/**
	 * 查询所有-分页
	 * @param pageNum 当前页面
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);

	/**
	 * 根据条件查询，分页
	 * @param typeTemplate
	 * @param pageNum 当前页面
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(TypeTemplate typeTemplate, int pageNum, int pageSize);

	/**
	 * 删除
	 * @return
	 */
	public void delete(Long[] ids);

}

File: mall-manager-web/src/main/webapp/js/service/typeTemplateService.js
/**
 * 类型模板服务层
 * Created by Wwl on 2019/3/4
 */
app.service('typeTemplateService',function ($http) {

    //增加
    this.add = function (entity) {
        return $http.post('../typeTemplate/add.do',entity);
    }

    //修改
    this.update = function (entity) {
        return $http.post('../typeTemplate/update.do',entity);
    }

    //删除
    this.delete = function (ids) {
        return $http.get('../typeTemplate/delete.do?ids='+ids);
    }

    //查询实体
    this.findOne = function (id) {
        return $http.get('../typeTemplate/findOne.do?id='+id);
    }

    //查询列表
    this.findList = function(){
        return $http.get('../typeTemplate/list.do');
    }

    //分页查询
    this.findPage = function(page, size){
        return $http.get('../typeTemplate/listByPage.do?page=' + page + '&size=' + size);
    }

    //搜索
    this.search = function (page,size,searchEntity) {
        return $http.post('../typeTemplate/search.do?page='+page+'&size='+size,searchEntity);
    }

});